src/backend/distributed/planner/function_call_delegation.c

```diff
@@ -168,9 +175,28 @@ TryToDelegateFunctionCall(DistributedPlanningContext *planContext)
 			{
 				RangeTblEntry *rtentry = rt_fetch(reference->rtindex,
 												  planContext->query->rtable);
-				if (rtentry->rtekind != RTE_RESULT)
+
+				if (rtentry->rtekind == RTE_FUNCTION &&
+					list_length(rtentry->functions) == 1)
+				{
+					RangeTblFunction *rtfunc = (RangeTblFunction *) linitial(
+						rtentry->functions);
+
+					if (IsA(rtfunc->funcexpr, FuncExpr))
+					{
+						/* function from the FROM clause e.g. SELECT * FROM fn() */
+						functionInFromList = true;
+						funcExpr = (FuncExpr *) rtfunc->funcexpr;
+					}
+					else
+					{
+						return NULL;
```

Comment:
should there be a debug message? (and a comment?)

Reply:
Simplified it. Maybe we can have a generic piece of code to fish out a simple function from FROM clause, we can reuse it here too **_IsMongoCollectionRTE_**()?

Reply:
Made an attempt to use some common code, but it's not ideal as requirements vary slightly.